### PR TITLE
Neutron  v2.0.0 upgrade doc NTRN-137

### PR DESCRIPTION
**Pull request by quasisamurai**
I'm gonna use this branch as a base for [this](https://hadronlabs.atlassian.net/browse/NTRN-166) task
so duality mention is missing due this fact

### CODE REVIEW

File: docs/neutron/v2-upgrade/overview.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+Neutron has transitioned from cosmos-sdk v0.45 to the more advanced v0.47, encompassing significant improvements and custom adaptations. The key aspects of this upgrade include:
+
+- **Primary Changes:**
+    - Reference to the main [cosmos-sdk v0.47 documentation](https://docs.cosmos.network/v0.47/learn/intro/overview).
+    - Access to the full changelog [here](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0/CHANGELOG.md).
+
+- **Neutron's Custom Fork:**
+    - We have created [our own fork](https://github.com/neutron-org/cosmos-sdk) of the cosmos-sdk, introducing unique enhancements tailored to our needs:
+        - **Gas Counting Exclusion:** Removal of gas counting in the upgrade module's begin blocker for more consistent gas accounting.
```
Comment (pr0n00gler) [inline docs/neutron/v2-upgrade/overview.md:15]: Please attach a relevant PR in our SDK here

File: docs/neutron/v2-upgrade/overview.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        - **BankHooks Introduction:** Implementation of [BankHooks](https://github.com/neutron-org/cosmos-sdk/pull/2), a pivotal feature for the new Tokenfactory.
+        - **Optimized Slashing Calculation:** Backporting of slashing missed blocks calculation from sdk v0.50.
```
Comment (pr0n00gler) [inline docs/neutron/v2-upgrade/overview.md:17]: Please attach a relevant PR in our SDK here

File: docs/neutron/v2-upgrade/overview.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        - **Module Parameters Handling:** Deprecation of the [x/params module](https://docs.cosmos.network/v0.47/modules/params). Modules now manage parameters directly.
+        - **IBC-Go Upgrade:** Moving to ibc-go v7 for improved inter-blockchain communication.
+        - **Technical Enhancements:** Several minor yet impactful technical improvements.
+
+## Wasmd Upgrade
+
+Our custom fork of [wasmd](https://github.com/neutron-org/wasmd), based on version 0.45, brings forward these key developments:
+
+- **Instantiate2 Activation:** Enabling predictable contract addresses through the [instantiate2 feature](https://docs.rs/cosmwasm-std/1.2.1/cosmwasm_std/fn.instantiate2_address.html).
+- **Smart-Contract Size Limit Increase:** Expansion of the binary size limit from 800KB to 3MB as [explained here](https://github.com/neutron-org/neutron/pull/320).
```
Comment (pr0n00gler) [inline docs/neutron/v2-upgrade/overview.md:29]:
```suggestion
- **Smart-Contract Size Limit Increase:** Expansion of the binary size limit from 800KB to 1.6MB as [explained here](https://github.com/neutron-org/neutron/pull/320).
```

File: docs/neutron/v2-upgrade/overview.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+- **Additional Improvements:** Various other technical modifications and advancements.
```
Comment (pr0n00gler) [inline docs/neutron/v2-upgrade/overview.md:30]: Please attach a link to wasmd 0.45 release with a changelog

File: docs/neutron/v2-upgrade/overview.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+- **Neutron's Custom Fork:**
+    - We have created [our own fork](https://github.com/neutron-org/cosmos-sdk) of the cosmos-sdk, introducing unique enhancements tailored to our needs:
+        - **Gas Counting Exclusion:** Removal of gas counting in the upgrade module's begin blocker for more consistent gas accounting.
+        - **BankHooks Introduction:** Implementation of [BankHooks](https://github.com/neutron-org/cosmos-sdk/pull/2), a pivotal feature for the new Tokenfactory.
+        - **Optimized Slashing Calculation:** Backporting of slashing missed blocks calculation from sdk v0.50.
+        - **CometBFT Transition:** A significant shift to CometBFT for enhanced consensus reliability.
+        - **ABCI 1.0 Support:** Enabling chains to implement their mempool with ABCI 1.0 compatibility.
+        - **Module Parameters Handling:** Deprecation of the [x/params module](https://docs.cosmos.network/v0.47/modules/params). Modules now manage parameters directly.
+        - **IBC-Go Upgrade:** Moving to ibc-go v7 for improved inter-blockchain communication.
+        - **Technical Enhancements:** Several minor yet impactful technical improvements.
```
Comment (pr0n00gler) [inline docs/neutron/v2-upgrade/overview.md:22]: Please attach a link to Cosmos SDK 0.47 release with a changelog

File: docs/neutron/v2-upgrade/overview.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+Enhancements within Neutron focus on integrating new modules, refining existing functionalities, and ensuring better alignment with the upgraded Cosmos SDK:
+
+### Globalfee Module Integration
+
+- **Minimum Gas Price Enforcement:** A mechanism implemented via the [globalfee module](https://github.com/cosmos/gaia/blob/feat/sdk-47-ibc-7/docs/modules/globalfee.md) to standardize gas prices across validators.
+
+### Tokenfactory Module Update
+
+- **BankHooks Activation:** Introduction of BankHooks for smart contracts handling token transfers, as detailed [here](https://github.com/neutron-org/neutron/blob/update-sdk47/x/tokenfactory/keeper/before_send.go).
```
Comment (pr0n00gler) [inline docs/neutron/v2-upgrade/overview.md:42]: Would be better to provide link to our documentation which describes these hooks (not the code)

File: docs/neutron/v2-upgrade/overview.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+- **Fee Removal for Token Creation:** Elimination of creation fees for Tokenfactory tokens, promoting free token generation on Neutron ([source](https://github.com/neutron-org/neutron/blob/e605ed3db4381994ee8185ba4a0ff0877d34e67f/app/upgrades/v2.0.0/upgrades.go#L157)).
+
+### Interchain Transactions and ContractManager Module Refactor
+
+- **ICA Usability Improvements:** Enhanced Interchain Account (ICA) functionality for a more user-friendly and secure experience for developers.
+- **Sudo Execution Error Handling:** Streamlined error message retrieval in the ContractManager module.
+- **New Fee Structure for ICA Creation:** Introduction of a fee system for developers creating ICAs on remote chains.
```
Comment (pr0n00gler) [inline docs/neutron/v2-upgrade/overview.md:49]: Please attach relevant PR's and links to our documentation

**Review by pr0n00gler — CHANGES_REQUESTED** [review]

File: docs/neutron/v2-upgrade/overview.md (hunk 5 above)
Comment (quasisamurai) [reply docs/neutron/v2-upgrade/overview.md:22]: it's already there, line `Access to the full changelog...`

File: docs/neutron/overview.md [hunk 8]
```diff
@@ -26,4 +26,64 @@ CosmWasm is a smart contracting platform built for the Cosmos ecosystem.
...
+
+<details> <summary> v2.0.0</summary>
+
+##### Overview
+
+This section provides a comprehensive overview of the significant changes and updates in Neutron version 2, focusing on the upgrades to Cosmos SDK, Wasmd, and Neutron's internal modules.
+
+#### Cosmos SDK Upgrade
+
+Neutron has transitioned from cosmos-sdk v0.45 to the more advanced v0.47, encompassing significant improvements and custom adaptations. The key aspects of this upgrade include:
```
Comment (pr0n00gler) [inline docs/neutron/overview.md:41]: In different places you say  `cosmos-sdk`, `sdk`, `Cosmos SDK`. Let's use `Cosmos SDK` everywhere, pls

File: docs/neutron/overview.md [hunk 9]
```diff
@@ -26,4 +26,64 @@ CosmWasm is a smart contracting platform built for the Cosmos ecosystem.
 
 ### F.A.Q.
 
-Please refer to the [F.A.Q.](./faq.md) page!
\ No newline at end of file
+Please refer to the [F.A.Q.](./faq.md) page!
+
+### Updates
```
Comment (pr0n00gler) [inline docs/neutron/overview.md:31]: Let's rename it `Releases`

File: docs/neutron/overview.md [hunk 10]
```diff
@@ -26,4 +26,64 @@ CosmWasm is a smart contracting platform built for the Cosmos ecosystem.
...
+
+<details> <summary> v2.0.0</summary>
```
Comment (pr0n00gler) [inline docs/neutron/overview.md:33]: `<details> <summary><b>v2.0.0</b></summary>`

Comment (pr0n00gler) [reply docs/neutron/overview.md:33]: Also we need a link to a release page on Github

**Review by pr0n00gler — CHANGES_REQUESTED** [review]
There is no mention of Duality integration in v2 release, pls add it

File: docs/neutron/overview.md [hunk 11]
```diff
@@ -26,4 +26,64 @@ CosmWasm is a smart contracting platform built for the Cosmos ecosystem.
...
+
+##### Overview
+
+This section provides a comprehensive overview of the significant changes and updates in Neutron version 2, focusing on the upgrades to Cosmos SDK, Wasmd, and Neutron's internal modules.
```
Comment (pr0n00gler) [inline docs/neutron/overview.md:37]: Let's put an overview outside a spoiler, so users could see a small description of the release.
And under the spoiler there will be a full context of a release

**Review by pr0n00gler — CHANGES_REQUESTED** [review]

File: docs/neutron/overview.md (hunk 8 above)
Comment (quasisamurai) [reply docs/neutron/overview.md:41]: Duality docs are gonna be merged in this repo as I mentioned in PR header

File: docs/neutron/neutron-core-releases.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+##### Tokenfactory Module Update
+
+- **BankHooks Activation:** Introduction of BankHooks for smart contracts handling token transfers, as detailed [here](neutron/modules/3rdparty/osmosis/ibc-hooks/overview).
+- **Fee Removal for Token Creation:** Elimination of creation fees for Tokenfactory tokens, promoting free token generation on Neutron ([source](https://github.com/neutron-org/neutron/blob/e605ed3db4381994ee8185ba4a0ff0877d34e67f/app/upgrades/v2.0.0/upgrades.go#L157)).
+
+##### Interchain Transactions and ContractManager Module Refactor
+
+- **ICA Usability Improvements:** Enhanced Interchain Account (ICA) functionality for a more user-friendly and secure experience for developers.
+- **Sudo Execution Error Handling:** Streamlined error message retrieval in the ContractManager module.
+- **New Fee Structure for ICA Creation:** [Introduction](https://github.com/neutron-org/neutron/pull/334) of a fee system for developers creating ICAs on remote chains. [Learn more](neutron/modules/interchain-txs/messages) (see Cosmos SDK v0.47 Update section)
```
Comment (pr0n00gler) [inline docs/neutron/neutron-core-releases.md:53]: **Learn more** link is 404

File: docs/neutron/neutron-core-releases.md [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+    - We have created [our own fork](https://github.com/neutron-org/cosmos-sdk) of the Cosmos SDK, introducing unique enhancements tailored to our needs:
+        - **Gas Counting Exclusion:** Removal of gas counting in the upgrade module's begin blocker for more consistent gas accounting.
+        - **BankHooks Introduction:** Implementation of [BankHooks](https://github.com/neutron-org/cosmos-sdk/pull/2), a pivotal feature for the new Tokenfactory.
+        - **[Optimized Slashing Calculation](https://github.com/neutron-org/cosmos-sdk/pull/5):** Backporting of slashing missed blocks calculation from Cosmos SDK v0.50.
+        - **CometBFT Transition:** A significant shift to CometBFT for enhanced consensus reliability.
+        - **ABCI 1.0 Support:** Enabling chains to implement their mempool with ABCI 1.0 compatibility.
+        - **Module Parameters Handling:** Deprecation of the [x/params module](https://docs.cosmos.network/v0.47/modules/params). Modules now manage parameters directly.
+        - **IBC-Go Upgrade:** Moving to ibc-go v7 for improved inter-blockchain communication.
+        - **Technical Enhancements:** Several minor yet impactful technical improvements.
```
Comment (pr0n00gler) [inline docs/neutron/neutron-core-releases.md:26]: These are no Neutron's Custom Fork changes. It should be moved to **Primate changes** section.

File: docs/neutron/neutron-core-releases.md [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+This section provides a comprehensive overview of the significant changes and updates in [Neutron version 2](https://github.com/neutron-org/neutron/releases/tag/v2.0.0), focusing on the upgrades to Cosmos SDK, Wasmd, and Neutron's internal modules.
+
+## v2.0.0
+
+#### Cosmos SDK Upgrade
+
+Neutron has transitioned from Cosmos SDK v0.45 to the more advanced v0.47, encompassing significant improvements and custom adaptations. The key aspects of this upgrade include:
+
+- **Primary Changes:**
+    - Reference to the main [Cosmos SDK v0.47 documentation](https://docs.cosmos.network/v0.47/learn/intro/overview).
```
Comment (pr0n00gler) [inline docs/neutron/neutron-core-releases.md:14]: Please add

-  `--broadcast-mode` block was removed. You need to query the result for a TX with neutrond q tx <hash> instead;

- the SDK version includes some key store migration for the CLI. Make sure you backup your private keys before testing this! You can not switch back to v0.45;

File: docs/neutron/neutron-core-releases.md [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+        - **IBC-Go Upgrade:** Moving to ibc-go v7 for improved inter-blockchain communication.
+        - **Technical Enhancements:** Several minor yet impactful technical improvements.
+
+#### Wasmd Upgrade
+
+Our custom fork of [wasmd](https://github.com/neutron-org/wasmd), based on version [0.45](https://github.com/CosmWasm/wasmd/blob/main/CHANGELOG.md), brings forward these key developments:
+
+- **Instantiate2 Activation:** Enabling predictable contract addresses through the [instantiate2 feature](https://docs.rs/cosmwasm-std/1.2.1/cosmwasm_std/fn.instantiate2_address.html).
+- **Smart-Contract Size Limit Increase:** Expansion of the binary size limit from 800KB to 1.6MB as [explained here](https://github.com/neutron-org/neutron/pull/320).
+- **Additional Improvements:** Various other technical modifications and advancements.
```
Comment (pr0n00gler) [inline docs/neutron/neutron-core-releases.md:34]: Please add:
- legacy REST endpoints for queries and txs are completely removed and only gRPC endpoints must be used now;

And make it bold, cause it's very important.

Also pls add:
- contracts are able to use floating point operations;

**Review by pr0n00gler — CHANGES_REQUESTED** [review]

File: docs/neutron/neutron-core-releases.md [hunk 16]
```diff
@@ -0,0 +1,58 @@
+# Neutron Core Releases
+
+## Overview
+
+This section provides a comprehensive overview of the significant changes and updates in [Neutron version 2](https://github.com/neutron-org/neutron/releases/tag/v2.0.0), focusing on the upgrades to Cosmos SDK, Wasmd, and Neutron's internal modules.
```
Comment (pr0n00gler) [inline docs/neutron/neutron-core-releases.md:5]: > of the significant changes and updates in Neutron version 2
Actually it does not anymore.

It provides info about all the releases of Neutron, not only about v2.

File: docs/neutron/neutron-core-releases.md [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+
+## v2.0.0
```
Comment (pr0n00gler) [inline docs/neutron/neutron-core-releases.md:7]: Let's make it a link to a Github Release page

**Review by pr0n00gler — CHANGES_REQUESTED** [review]

**Review by pr0n00gler — APPROVED** [review]
LGTM, waiting for DEX docs merge

**Pull request merged**